test/index.html

```diff
@@ -0,0 +1,250 @@
+
+        if( nextStep[i].next ) {
+          assert.ok( nextStep[i].next(), nextStep[i].text );
+          i++;
+        } else {
+          // Remember to cleanup, since we're operating outside of qunit-fixture
+          root.removeEventListener( "impress:stepenter", registeredListener );
+          doneWithNext();
+        }
+      };
+      
+      root.addEventListener( "impress:stepenter", function( event ) {
+        var registeredListener = this;
+        setTimeout( function() { assertStepEnter( event, registeredListener ) },
+                    wait*2 ); // wait*2 ensures that assertStepLeave is finished 
+                              // first, since here we'll call nextStep.next and i++
+      });
+
+      var prevStep = [ step1 ];
+      for(j=0; j < nextStep.length; j++){
+        prevStep.push( nextStep[j].me );
+      }
+      var assertStepLeave = function( event, registeredListener ) {
+        assert.equal( event.target, prevStep[i],
+                      event.target.id + " triggered impress:stepleave event." );
+        assert.ok( !event.target.classList.contains("active"),
+                   event.target.id + " unset active css class." );
+        assert.ok( !event.target.classList.contains("present"),
+                   event.target.id + " unset present css class." );
+        assert.ok( !event.target.classList.contains("future"),
+                   event.target.id + " unset future css class." );
+        assert.ok( event.target.classList.contains("past"),
+                   event.target.id + " set past css class." );
+        
+        if( i+2 == prevStep.length ) {
+          root.removeEventListener( "impress:stepleave", registeredListener );
+          doneWithPrev();
+        }
+
+      };
+
+      root.addEventListener( "impress:stepleave", function( event ) {
+        var registeredListener = this;
+        setTimeout( function() { assertStepLeave( event, registeredListener ) },
+                    wait );
+      });
+
+      // Do other tests first, then trigger the chain reaction we setup above.
+      assert.deepEqual( impress().goto(document.querySelector( "div#impress" )),
+                        false,
+                        "goto() to a non-step element fails, as it should." );
+      assert.deepEqual( impress().goto(),
+                        false,
+                        "goto(<nothing>) fails, as it should." );
+
+      // This starts executing the chain of event listeners above     
+      assert.ok( impress().next(),
+                 "next() called and returns ok (1->2)" );
+    });
+    
+
+
+
+
+
+    // Note, it's clearly implied in the impress.js code, that when we support
+    // plugins, the keypress api should be a plugin. Then this test of course
+    // should go there as well.
+    QUnit.test( "Keypress API", function( assert ) {
+      var done = assert.async();
+
+    });
+
+    // Cleanup
+    QUnit.test( "Cleanup", function( assert ) {
```

Comment:
Do you mean the beforeEach()/afterEach() functionality that is part of QUnit.module(). I don't think that applies here. Or it could, but depends on what the other tests do and what they need to clean up. For now, I'm assuming each test() block will do its own cleanup.